docs/running-on-kubernetes.md

```diff
@@ -529,8 +600,11 @@ specific to Spark on Kubernetes.
   <td><code>spark.kubernetes.driver.pod.name</code></td>
   <td>(none)</td>
   <td>
-    Name of the driver pod. If not set, the driver pod name is set to "spark.app.name" suffixed by the current timestamp
-    to avoid name conflicts.
+    Name of the driver pod. In cluster mode, if this is not set, the driver pod name is set to "spark.app.name"
+    suffixed by the current timestamp to avoid name conflicts. In client mode, if your application is running
+    inside a pod, it is highly recommended to set this to the name of the pod your driver is running in. Setting this
+    value in client mode allows the driver to inform the cluster that your application's executor pods should be
```

Comment:
Instead of saying `inform the cluster that your application's executor pods should be deleted ...`, it's better just say `to allows the driver pod to become the owner of your application's executor pods for garbage collection to work`.